### PR TITLE
Point script query docs at runtime fields

### PR DESCRIPTION
**Pull request by nik9000**
This adds a "note" on the docs for the script query pointing folks to
runtime fields because they are more flexible. It also translates the
request example into runtime fields.

Relates to #69291

### CODE REVIEW

File: docs/reference/query-dsl/script-query.asciidoc [hunk 1]
```diff
@@ -4,6 +4,12 @@
 <titleabbrev>Script</titleabbrev>
 ++++
 
+NOTE: <<runtime,Runtime fields>> provide a very similar feature that is quite
+      a bit more flexible. You write a script to create field values and they
+      are available everywhere - <<search-fields, `fields`>>,
+      <<query-dsl, all queries>>, <<search-aggregations, aggregations>>,
+      and anything else you can think of.
```
Comment (lockewritesdocs) [inline docs/reference/query-dsl/script-query.asciidoc:11]:
```suggestion
NOTE: <<runtime,Runtime fields>> provide a very similar feature that is
      more flexible. You write a script to create field values and they
      are available everywhere, such as <<search-fields, `fields`>>,
      <<query-dsl, all queries>>, and <<search-aggregations, aggregations>>.

```

**Review by lockewritesdocs — APPROVED** [review]
Thanks for adding this note @nik9000! Left a suggested, non-blocking change.

**Comment by nik9000** [comment]
> Thanks for adding this note @nik9000! Left a suggested, non-blocking change.

Sounds good! Thanks.

**Review by javanna — COMMENTED** [review]
I wonder if it makes sense to come up with a plan around deprecating and removing the script query and similar features that can be entirely replaced by runtime fields. Possibly part of this plan would be to not document them anymore. In this page it is odd that we even spend time documenting script query, to then get to a superior solution that replaces it. I believe we could have already deprecated it at least?

**Comment by nik9000** [comment]
Yeah.

**Comment by lockewritesdocs** [comment]
If we plan to deprecate the Script query, we can add a similar deprecation notice like we did for [Kibana scripted fields](https://www.elastic.co/guide/en/kibana/7.x/scripted-fields.html). We would need buy-in from PM I believe before adding the deprecation notice.

**Comment by nik9000** [comment]
Deprecating `script-query` also means sending back deprecation notices over the API. It's probably a good thing.

**Pull request merged**